### PR TITLE
Added specific warning for build in and custum icons

### LINKED ISSUE #1470: [Enhancement] Better warnings for undefined/missing icons
**Please add a short description of your enhancement request**
When the Icon component can't find an icon it logs a warning to the console, but doesn't log whether it's a built-in icon (`<kirby-icon name="built-in-icon-name" ...>`) or a custom icon (`<kirby-icon customName="custom-icon-name" ...>`). This makes debugging harder.
See: https://github.com/kirbydesign/designsystem/blob/master/libs/designsystem/src/lib/components/icon/icon.component.ts#L27

**Describe the solution you'd like**
I would like for the icon component to log different warning messages based on whether the missing icon is built-in or custom. For built-in icons the error is related to a wrong icon name (or a typo), for custom icons the error can also be related to a wrong icon name (or a typo) but also a mis-configured custom icon registration.
**Suggested warnings:** 
* Built-in (`name="..."`): `Built-in icon with name "${this.name}" was not found. Do you have a typo in 'name' or did you mean to use a custom icon? If so, please use: <kirby-icon customName="${this.name}"></kirby-icon>`.
* Custom (`customName="..."`): `Custom icon with name "${this.customName}" was not found. Do you have a typo in 'customName' or forgot to configure the custom icon through the 'IconRegistryService'?`.

### Kick Off:

- [X] Ensure the enhancement is `UX refined` and aligned with UX 
_The component and/or enhancement should be published and available in the [Kirby Styleguide on Zeplin](https://zpl.io/258pXGj)_
- [X] Ensure the enhancement has been `Tech refined` with @kirbydesign/kirby-guild and this issue is updated with a clear implementation description  
_This issue should be in the [Ready to do](https://github.com/kirbydesign/designsystem/projects/1#column-4590936) column of the [Kirby kan-ban board](https://github.com/kirbydesign/designsystem/projects/1) before starting implementation)_
- [x] Assign yourself to this issue and move it to the [In progress](https://github.com/kirbydesign/designsystem/projects/1#column-4590937) column of the [Kirby kan-ban board](https://github.com/kirbydesign/designsystem/projects/1)

### Code:

- [x] Create Feature Branch from [master branch](https://github.com/kirbydesign/designsystem/tree/master)
- [x] Create a draft implementation and push to Github
- [x] Ask a member of @kirbydesign/kirby-guild for a WIP review by creating a draft Pull Request
- [x] Implement unit tests
- [ ] Update Cookbook Examples and Showcase, i.e. see [Radio](https://cookbook.kirby.design/home/showcase/radio)  
_Also remember to add any relevant new API documentation_

### Review:

- UX review:
  - [x] <s>Ensure implementation is correct in relation to the UX design and the [Kirby Styleguide on Zeplin](https://zpl.io/258pXGj)</s>
  - [x] <s>With UX agree on the version of the implementation</s>
- Code review:
  - [ ] Open a pull request (or mark the existing draft PR as `Ready for review`) and ask @kirbydesign/kirby-guild for a review  
_Remember to add `closes #issueno` to the description of the PR._
  - [ ] Once approved, merge feature branch/PR to master
- [ ] Ask a member of @kirbydesign/kirby-guild to add a link to component showcase from Kirby Component Status and update the version number

:tada: Celebrate

### PR DESCRIPTION
**Pull request by bdytos**
This PR closes #1470 

## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [x] Enhancement (to existing content)
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] cookbook application / infrastructure changes
- [ ] Other... Please describe:

## Which issue documents the current behaviour?

Issue Number: #[replace with issue number]

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: libs/designsystem/src/lib/components/icon/icon.component.ts [hunk 1]
```diff
@@ -50,6 +50,18 @@ export class IconComponent implements OnChanges {
     }
   }
 
+  private showWarning(): string {
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/icon/icon.component.ts:53]: Consider naming:
```suggestion
  private getMissingIconWarning(): string {
```

File: libs/designsystem/src/lib/components/icon/icon.component.ts [hunk 2]
```diff
@@ -34,7 +34,7 @@ export class IconComponent implements OnChanges {
   set icon(icon: Icon) {
     // If icon are not found, set default icon
     if (!icon && (this.name || this.customName)) {
-      console.warn(`Icon with name "${this.name || this.customName}" was not found.`);
+      console.warn(this.showWarning());
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/icon/icon.component.ts:37]: Consider either renaming method for getting warning message or move `console.warn` to warning method.

File: libs/designsystem/src/lib/components/icon/icon.component.ts [hunk 3]
```diff
@@ -50,6 +50,18 @@ export class IconComponent implements OnChanges {
...
+
+    if (this.customName) {
+      warning = `Custom icon with name "${this.customName}" was not found. Do you have a typo in 'customName' or forgot to configure the custom icon through the 'IconRegistryService'?`;
+    } else {
+      warning = `Built-in icon with name "${this.name}" was not found. Do you have a typo in 'name' or did you mean to use a custom icon? If so, please use: <kirby-icon customName="${this.name}"></kirby-icon>`;
+    }
+
+    return warning;
+  }
+
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/icon/icon.component.ts:64]: Consider readability:
```suggestion
    if (this.customName) {
      return `Custom icon with name "${this.customName}" was not found. Do you have a typo in 'customName' or forgot to configure the custom icon through the 'IconRegistryService'?`;
    }
      return `Built-in icon with name "${this.name}" was not found. Do you have a typo in 'name' or did you mean to use a custom icon? If so, please use: <kirby-icon customName="${this.name}"></kirby-icon>`;
  }

```

File: libs/designsystem/src/lib/components/icon/icon.component.ts (hunk 1 above)
Comment (bdytos) [reply libs/designsystem/src/lib/components/icon/icon.component.ts:53]: Renamed function name

File: libs/designsystem/src/lib/components/icon/icon.component.ts (hunk 2 above)
Comment (bdytos) [reply libs/designsystem/src/lib/components/icon/icon.component.ts:37]: Renamed method

File: libs/designsystem/src/lib/components/icon/icon.component.ts (hunk 3 above)
Comment (bdytos) [reply libs/designsystem/src/lib/components/icon/icon.component.ts:64]: Added "line-break" to template string for readability

File: libs/designsystem/src/lib/components/icon/icon.component.ts [hunk 4]
```diff
@@ -34,7 +34,8 @@ export class IconComponent implements OnChanges {
   set icon(icon: Icon) {
     // If icon are not found, set default icon
     if (!icon && (this.name || this.customName)) {
-      console.warn(this.getMissingIconWarning());
+      this.getMissingIconWarning();
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/icon/icon.component.ts:37]: Consider naming after moving the actual `console.warn` to method ;)

**Review by jakobe — CHANGES_REQUESTED** [review]

File: libs/designsystem/src/lib/components/icon/icon.component.ts [hunk 5]
```diff
@@ -50,6 +51,19 @@ export class IconComponent implements OnChanges {
     }
   }
 
+  private getConsoleWarningsForMissingIcons(): void {
```
Comment (jakobe) [inline libs/designsystem/src/lib/components/icon/icon.component.ts:54]: Consider naming (this method doesn't "get" anything)

**Review by jakobe — CHANGES_REQUESTED** [review]

**Review by jakobe — APPROVED** [review]

**Pull request merged**